frame/examples/kitchensink/src/tests.rs

```diff
@@ -0,0 +1,108 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Tests for pallet-example-kitchensink.
+
+use crate::*;
+use frame_support::{traits::ConstU64, parameter_types};
+use sp_runtime::BuildStorage;
+use frame_support::macro_magic::use_attr;
+#[use_attr]
+use frame_support::derive_impl;
+// Reexport crate as its pallet name for construct_runtime.
+use crate as pallet_example_kitchensink;
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+type Block = frame_system::mocking::MockBlock<Test>;
+
+// For testing the pallet, we construct a mock runtime.
+frame_support::construct_runtime!(
+	pub enum Test where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+		Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>},
+		Kitchensink: pallet_example_kitchensink::{Pallet, Call, Storage, Config<T>, Event<T>},
+	}
+);
+
+/// Using a default config for [`frame_system`] in tests. See `default-config` example for more 
```

Comment:
Perhaps it is worth bringing this in as a dependency for linking purposes?

Reply:
Do you mean to add the `default-config` example as a dependency in the `kitchensink` example?

Reply:
Tests are not documented AFAIK.